pystac_client/item_search.py

```diff
@@ -590,71 +590,24 @@ def matched(self) -> Optional[int]:
             warnings.warn("numberMatched or context.matched not in response")
         return found
 
-    def get_item_collections(self) -> Iterator[ItemCollection]:
-        """DEPRECATED. Use :meth:`ItemSearch.item_collections` instead.
-
-        Yields:
-            ItemCollection : a group of Items matching the search criteria within an
-            ItemCollection
-        """
-        warnings.warn(
-            "get_item_collections() is deprecated, use item_collections() instead",
-            DeprecationWarning,
-        )
-        return self.item_collections()
-
-    def item_collections(self) -> Iterator[ItemCollection]:
-        """Iterator that yields ItemCollection objects.  Each ItemCollection is
-        a page of results from the search.
-
-        Yields:
-            ItemCollection : a group of Items matching the search criteria within an
-            ItemCollection
-        """
-        if isinstance(self._stac_io, StacApiIO):
-            for page in self._stac_io.get_pages(
-                self.url, self.method, self.get_parameters()
-            ):
-                yield ItemCollection.from_dict(
-                    page, preserve_dict=False, root=self.client
-                )
-
-    def get_items(self) -> Iterator[Item]:
-        """DEPRECATED. Use :meth:`ItemSearch.items` instead.
-
-        Yields:
-            Item : each Item matching the search criteria
-        """
-        warnings.warn(
-            "get_items() is deprecated, use items() instead",
-            DeprecationWarning,
-        )
-        return self.items()
+    # ------------------------------------------------------------------------
```

Comment:
Lots of movement here, just to group together related methods. In general things are grouped by result set (item, page, everything) with pystac followed by plain.

All the deprecated methods are at the bottom.